core/src/processing/javafx/PSurfaceFX.java

```diff
@@ -706,17 +706,82 @@ protected void fxKeyEvent(javafx.scene.input.KeyEvent fxEvent) {
     }
 
     long when = System.currentTimeMillis();
-    KeyCode kc = fxEvent.getCode();
-    // Are they f*ing serious?
-    char key;
-    if (et == KeyEvent.KEY_TYPED) {
-      key = fxEvent.getCharacter().charAt(0);
```

Comment:
I did get an `ArrayIndexOutOfBoundsException` on this line earlier today (I think I hit ESC to close a sketch), in case that's not fixed in this commit.